### DISCUSSION TITLE
Streaming- offset- specifying message so it reads from it

### DISCUSSION THREAD

**Discussion by suprunka** [rabbitmq/rabbitmq-server#3377] [Other]
Is it possible to specify message somehow, so the stream is not read from first/last/next/offset, but the message with some property. I would like to have a property on message that would specify the stream start reading from this. Would it be possible, othewise how x-stream-offset for timestam in .net works. Can I read only message that have been published after that timestamp, because I feel it reads the timestamp as offset and it says that 'Value of type 'DateTime' cannot appear as table value'.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by acogoluegnes** [comment]
It is not possible to start reading after some given message, but using a timestamp should work.

**Comment by acogoluegnes** [comment]
You can also have a look at the following mailing list post: https://groups.google.com/g/rabbitmq-users/c/P6QAZgbvlUc/m/MoVV-VaSBAAJ.

**Comment by kjnilsson** [comment]
I think there is an issue in the .NET client which means you cannot set a consumer arg as a timestamp but I haven't yet had time to look at why that is or if there is a workaround. For now you could use a time specifier string such as "5m" (5 minutes ago) or 1d (1 day ago).
